templates/global/chart.html

```diff
@@ -1,96 +1,124 @@
-            {%for amount in amounts%} randomColor(),{%endfor%}];
+      const ctx = document.getElementById("right_chart").getContext("2d");
+      const randomColors = [
+          {%for amount in amounts%} randomColor(),{%endfor%}];
 
-        new Chart(ctx,
-            {
-                type: "pie",
-                data: {
-                    labels: [{%for category in categories%}
-                        '{{category}}',
-                    {%endfor%}],
-                        datasets: [
-                        {
-                            label: "Avg Amount",
-                            data: [{%for amount in amounts%}
-                                '{{amount}}',
-                            {%endfor%}],
-                            backgroundColor: randomColors,
-                            borderColor: randomColors,
-                            borderWidth: 1,
-                        },
-                    ],
-                },
-                options: {
-                    responsive: true,
-                    maintainAspectRatio: false,
-                    scales: {
-                        y: {
-                            beginAtZero: true,
-                        },
-                    },
-                    title: {
-                        display: true,
-                        text: 'Average expense amount per category',
+      new Chart(ctx,
+          {
+              type: "pie",
+              data: {
+                  labels: [{%for category in categories%}
+                      '{{category}}',
+                  {%endfor%}],
+                      datasets: [
+                      {
+                          data: [{%for amount in amounts%}
+                              '{{amount}}',
+                          {%endfor%}],
+                          backgroundColor: randomColors,
+                          borderColor: randomColors,
+                          borderWidth: 1,
+                      },
+                  ],
+              },
+              options: {
+                tooltips: {
+                    callbacks: {
+                        label: function(tooltipItem, data) {
+                            var dataset = data.datasets[tooltipItem.datasetIndex];
+                            var total = dataset.data.reduce(function(previousValue, currentValue, currentIndex, array) {
+                                return parseInt(previousValue) + parseInt(currentValue);
+                            });
+                            console.log(total)
+                            var currentValue = dataset.data[tooltipItem.index];
+                            var percentage = Math.floor(((currentValue/total) * 100)+0.5);
+
+                            return 'Expense Monthly Amount of ' + data.labels[tooltipItem.index] + ': ' + data.datasets[tooltipItem.datasetIndex].data[tooltipItem.index] + "$ which is " + percentage + "% percentage";
+                       }
```

Comment:
If I am not mistaken @svatares mentioned that the logic shouldn't be implemented here. it should be written as a helper and called here.

Reply:
In general code that has even the tiny bit of logic shouldn't be in an HTML file.
I haven't done that with django or python, but in the past I worked with both Angular and React and we never left code in the HTML files.
It can't be tested and can cause a lot of issues when bugs appears

Reply:
I am not sure I understand why. 
It seems like it is possible: https://openfolder.sh/django-keeping-logic-out-of-templates-and-views

Reply:
Lets book a meeting on the topic and hear each others thoughts on the subject. I think it doesn't have anything to do with this PR so Im resolving this conversation and we'll continue on zoom or slack